### PR TITLE
Prometheus: Tweak query builder select component

### LINKED ISSUE #56943: Inconsistent components / UIs for AdHoc filters & query builder. Also can't clear AdHoc filters
**What happened**:
When I use Ad-Hoc filters on a dashboard, they look like this:

image

But then if I:
- Explore one of the Panels (one which is backed by a Prometheus datasource)
- Then switch the query editor from "Code" to "Builder" mode

I then see this:
image

image

image

The visual appearance is very different, and the behaviour is also very different. IMO these should both look the same and behave the same.

**What you expected to happen**:
- The visuals should be the same, or at least more similar.
   - The orange and black used on the dashboard should also be used in the query-builder.
   - Also the outlines and down-arrows which always show on the query-builder are somewhat old-fashioned and a little ugly/busy on the eye perhaps, if I'm allowed to have an opinion on style (as a software engineer)!
- The front-end behaviour should be the same
   - The `X` to remove should be present on the Dashboard Ad-Hoc filters too. It's almost impossible to remove a dashboard Ad-Hoc filter. You have to know to click on the label-name, and then click the option "-- remove filter --" at the top of this list. But this just looks like another label-name (and what if I actually had a label named that way 🤯 ).
   - The available operators should be the same. Currently `>` and `<` are available only on the dashboard, not on the query-builder.
   - Other wrinkles - on the builder I can attempt to enter the label value before the key, or add a second filter before I've finished the first filter - neither of these are possible on the dashboard
- The back-end / data available in the picker should be the same
   - On the dashboard `org_id` is available to me in the list of label-keys. But in the query-builder `cluster, group, id` are my only options.

**Given the number of differences here, this might actually be 2 or more separate bugs**
Please feel free to split this ticket up as you see fit.

**How to reproduce it (as minimally and precisely as possible)**:
Create a dashboard with a Panel with a Prometheus datasource. Add an ad-hoc filter to the dashboard as a dashboard variable. Then start filtering, and repeat the steps described above.

**Anything else we need to know?**:
This discussion is relevant to the Cardinality datasource, where we are also in the process of implementing AdHoc filters on dashboards, and figuring out what the behaviour should be when moving to Explore mode. cc @darrenjaneczek 

**Environment**:
- Grafana version: 9.3.0-37457
- Data source type & version: Prometheus
- OS Grafana is installed on: GrafanaCloud
- User OS & Browser: Up to date MacOS, Chrome

### PR DESCRIPTION
**Pull request by itsmylife**
Fixes https://github.com/grafana/grafana/issues/56943

### CODE REVIEW

File: public/app/plugins/datasource/prometheus/querybuilder/components/LabelFilterItem.tsx [hunk 1]
```diff
@@ -6,9 +6,10 @@ import { selectors } from '@grafana/e2e-selectors';
 import { AccessoryButton, InputGroup } from '@grafana/experimental';
 import { AsyncSelect, Select } from '@grafana/ui';
 
-import { PROMETHEUS_QUERY_BUILDER_MAX_RESULTS } from '../components/MetricSelect';
 import { QueryBuilderLabelFilter } from '../shared/types';
 
+import { PROMETHEUS_QUERY_BUILDER_MAX_RESULTS } from './MetricSelect';
```
Comment (gtk-grafana) [inline public/app/plugins/datasource/prometheus/querybuilder/components/LabelFilterItem.tsx:11]: imported but never used?

File: public/app/plugins/datasource/prometheus/querybuilder/components/LabelFilterItem.tsx [hunk 2]
```diff
@@ -159,8 +161,10 @@ export function LabelFilterItem({
 }
 
 const operators = [
-  { label: '=~', value: '=~', isMultiValue: true },
   { label: '=', value: '=', isMultiValue: false },
   { label: '!=', value: '!=', isMultiValue: false },
+  { label: '<', value: '<', isMultiValue: false },
+  { label: '>', value: '>', isMultiValue: false },
```
Comment (gtk-grafana) [inline public/app/plugins/datasource/prometheus/querybuilder/components/LabelFilterItem.tsx:167]: This is pretty awesome new functionality!

File: public/app/plugins/datasource/prometheus/querybuilder/components/LabelFilterItem.tsx (hunk 1 above)
Comment (itsmylife) [reply public/app/plugins/datasource/prometheus/querybuilder/components/LabelFilterItem.tsx:11]: It is being used. See: https://github.com/grafana/grafana/pull/59749/files#diff-6ec019fb24e14b099249202ed75e8e378f6c68890de1abb139fef56e82ad70f8R124

Comment (gtk-grafana) [reply public/app/plugins/datasource/prometheus/querybuilder/components/LabelFilterItem.tsx:11]: Oh, it was just moved, sorry, I'm blind!

**Review by gtk-grafana — APPROVED** [review]

**Pull request merged**